src/union.ts

```diff
@@ -254,9 +253,16 @@ export class Union {
                 }
             }
         }
-        return Array.from(result).sort();
+        return Array.from(result.values());
     };
 
+    private pathFromReaddirEntry = (readdirEntry: readdirEntry): string => {
+        if (readdirEntry instanceof Buffer || typeof readdirEntry === 'string') {
+            return String(readdirEntry);
+        } else {
```

Comment:
else is never necessary: since you're returning in the truey branch, you can drop this else.